test/common/http/http2/codec_impl_test.cc

```diff
@@ -815,15 +828,24 @@ INSTANTIATE_TEST_CASE_P(Http2CodecImplTestDefaultSettings, Http2CodecImplTest,
                         ::testing::Combine(HTTP2SETTINGS_DEFAULT_COMBINE,
                                            HTTP2SETTINGS_DEFAULT_COMBINE));
 
+namespace {
+// Randomly picks a testing value to reduce the number of tests to run.
```

Comment:
Generally not a fan of random tests. What's the reason for this change?

Reply:
The test hits memory limit when runs in Forge (an internal system to run tests). But it runs fine with bazel. Maybe that means I can make the change in google3 only (?). I hoped the continuous build can cover all the settings combinations.

Currently, each test runs 255 times with different settings. Using randomly selected values, each test runs once. But I am totally OK to revert this change, :)

Reply:
@soya3129 can you chat with @htuch and @alyssawilk about the internal Google issue and circle back here?

Reply:
I think the main thing is we don't need to test with the Cartesian product, but can pick a baseline configuration and then do sensitivity analysis in the tests one parameter at a time. I would do that for the tests in this file, and if we one day we find a regression with more complex parameters, we can consider making this more complicated.

Reply:
Agreed. I still don't understand why this fails internally due to memory issues, that doesn't make any sense to me. But I probably don't want to know.

Mainly, I don't like random. Can we do min/max/mid testing or something like that across all the settings? That should greatly reduce the number of tests and allow us to revert the rest of these changs?

Reply:
OK! Will create another PR to reduce cases by running min/max/mid testing. Thanks for the reviews and discussion!